go/vt/vtgate/planbuilder/testdata/reference_cases.json

```diff
@@ -0,0 +1,644 @@
+        "Table": "ambiguous_ref_with_source",
+        "Variant": "Reference"
+      },
+      "Original": "select r1.col from ambiguous_ref_with_source r1 join ambiguous_ref_with_source",
+      "QueryType": "SELECT"
+    },
+    "gen4-plan": {
+      "Instructions": {
+        "FieldQuery": "select r1.col from ambiguous_ref_with_source as r1, ambiguous_ref_with_source where 1 != 1",
+        "Keyspace": {
+          "Name": "main",
+          "Sharded": false
+        },
+        "OperatorType": "Route",
+        "Query": "select r1.col from ambiguous_ref_with_source as r1, ambiguous_ref_with_source",
+        "Table": "ambiguous_ref_with_source",
+        "Variant": "Reference"
+      },
+      "Original": "select r1.col from ambiguous_ref_with_source r1 join ambiguous_ref_with_source",
+      "QueryType": "SELECT",
+      "TablesUsed": [
+        "main.ambiguous_ref_with_source"
+      ]
+    }
+  },
+  {
+    "comment": "ambiguous unqualified reference table can merge with other opcodes left to right.",
+    "query": "select ambiguous_ref_with_source.col from ambiguous_ref_with_source join user",
+    "v3-plan": {
+      "Instructions": {
+        "Inputs": [
+          {
+            "FieldQuery": "select ambiguous_ref_with_source.col from ambiguous_ref_with_source where 1 != 1",
+            "Keyspace": {
+              "Name": "main",
+              "Sharded": false
+            },
+            "OperatorType": "Route",
+            "Query": "select ambiguous_ref_with_source.col from ambiguous_ref_with_source",
+            "Table": "ambiguous_ref_with_source",
+            "Variant": "Reference"
+          },
+          {
+            "FieldQuery": "select 1 from `user` where 1 != 1",
+            "Keyspace": {
+              "Name": "user",
+              "Sharded": true
+            },
+            "OperatorType": "Route",
+            "Query": "select 1 from `user`",
+            "Table": "`user`",
+            "Variant": "Scatter"
+          }
+        ],
+        "JoinColumnIndexes": "L:0",
+        "OperatorType": "Join",
+        "TableName": "ambiguous_ref_with_source_`user`",
+        "Variant": "Join"
+      },
+      "Original": "select ambiguous_ref_with_source.col from ambiguous_ref_with_source join user",
+      "QueryType": "SELECT"
+    },
+    "gen4-plan": {
+      "Instructions": {
+        "FieldQuery": "select ambiguous_ref_with_source.col from ambiguous_ref_with_source, `user` where 1 != 1",
+        "Keyspace": {
+          "Name": "user",
+          "Sharded": true
+        },
+        "OperatorType": "Route",
+        "Query": "select ambiguous_ref_with_source.col from ambiguous_ref_with_source, `user`",
+        "Table": "`user`, ambiguous_ref_with_source",
+        "Variant": "Scatter"
+      },
+      "Original": "select ambiguous_ref_with_source.col from ambiguous_ref_with_source join user",
+      "QueryType": "SELECT",
+      "TablesUsed": [
+        "main.ambiguous_ref_with_source",
```

Comment:
The instruction seems right, but not sure this is?

Reply:
looks correct to me - these are the two tables you are querying, right?

Reply:
The Route looks good, but the TablesUsed should just have `ambiguous_ref_with_source`.
`main.ambiguous_ref_with_source` does not look correct.

Reply:
I think you are wrong, @harshit-gangal. We always add the keyspace prefix to table names in the TablesUsed field. We should never have a bare table name there

Reply:
I mean it should say it is going to `user`.`ambiguous_ref_with_source` and not `main`.`ambiguous_ref_with_source`